packages/template-retail-react-app/scripts/build-translations.js

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env node
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint @typescript-eslint/no-var-requires: "off" */
+
+/**
+ * This script will extract messages from base template and extended app and output all translations in the extended app
+ * If a file is overridden, it won't extract messages from that file in the base template
+ */
+const {exec} = require('child_process')
+const fs = require('fs')
+const path = require('path')
+const packagePath = path.join(process.cwd(), '.', 'package.json')
+const pkgJSON = JSON.parse(fs.readFileSync(packagePath))
+
+const getAllFilesByExtensions = (dirPath, arrayOfFiles = [], extensions = []) => {
+    const files = fs.readdirSync(dirPath)
+
+    files.forEach(function (file) {
+        if (fs.statSync(path.join(dirPath, file)).isDirectory()) {
+            arrayOfFiles = getAllFilesByExtensions(
+                path.join(dirPath, file),
+                arrayOfFiles,
+                extensions
+            )
+        } else {
+            arrayOfFiles.push(path.join(dirPath, file))
+        }
+    })
+    if (extensions.length) {
+        return arrayOfFiles.filter((filePath) => {
+            const getExtension = path.extname(filePath).replace('.', '')
+            return extensions.includes(getExtension)
+        })
+    }
+
+    return arrayOfFiles
+}
+
+try {
+    const isBaseTemplate = !!pkgJSON.ccExtensibility.extendable
```

Comment:
There's another kind of template to consider. We should also consider a regular template that does not have `extendable`. Currently, not all of our templates have it.

I'm thinking that it's better if the logic goes like this:
If isOverrides, ...
Else ...